### PR TITLE
Core get api

### PR DESCRIPTION
**Pull request by ginnytruong**

### CODE REVIEW

File: __tests__/app.tests.js [hunk 1]
```diff
@@ -33,4 +35,16 @@ describe("GET /api/invalidpath", () => {
             expect(response.body.msg).toBe("Not found");
         });
         });
-    });
\ No newline at end of file
+    });
+
+describe.only("GET /api", () => {
```
Comment (MKD01) [inline __tests__/app.tests.js]: Please dont forget to remove .only's so we can have some confidence in knowing all tests are passing.

File: __tests__/app.tests.js [hunk 2]
```diff
@@ -33,4 +35,16 @@ describe("GET /api/invalidpath", () => {
...
+    test("returns descriptions from endpoints.js for all api endpoints", () => {
+        return request(app)
+        .get("/api")
+        .expect(200)
+        .then((response) => {
+            const body = response.body;
+            expect(body).toEqual(endpoints);
+        })
+    })
```
Comment (MKD01) [inline __tests__/app.tests.js:49]: Really nice and simple test.

File: db/controller.js [hunk 3]
```diff
@@ -8,4 +8,14 @@ exports.getAllTopics = (request, response, next) => {
...
+};
+
+exports.getAllEndPoints = (request, response, next) => {
+    selectAllEndPoints()
+    .then((allEndPoints) => {
+        response.status(200).send(allEndPoints)
+    })
+    .catch((err) => {
+        next(err);
+    });
```
Comment (MKD01) [inline db/controller.js:20]: Nice to see a .catch incase anything goes wrong outside of our expectations.

File: db/model.js [hunk 4]
```diff
@@ -10,4 +12,11 @@ exports.selectAllTopics = () => {
...
     });
-};
\ No newline at end of file
+};
+
+exports.selectAllEndPoints = () => {
+    return fs.readFile("endpoints.json", "utf-8")
+    .then((data) => {
+        return JSON.parse(data);
+    });
```
Comment (MKD01) [inline db/model.js:21]: This definitely works, but dont forget you can also directly require in the endpoints.json file as well so you dont have to use the fs library for this.

**Review by MKD01 — COMMENTED** [review]
Really good work 👍

File: db/model.js (hunk 4 above)
Comment (ginnytruong) [reply db/model.js:21]: I see! I've updated this in the code now :) 
https://github.com/ginnytruong/nc-news/pull/5

**Pull request merged**